OrangeHRMTestFramework/Tests/OrangeHRM/EmployeeManagementTests.cs

```diff
@@ -0,0 +1,243 @@
+﻿using NUnit.Framework;
+using OrangeHRMTestFramework.Common.Drivers;
+using OrangeHRMTestFramework.Data;
+using OrangeHRMTestFramework.Data.Constants;
+using OrangeHRMTestFramework.Helpers;
+using OrangeHRMTestFramework.HttpClients;
+using OrangeHRMTestFramework.Models;
+using OrangeHRMTestFramework.PageObjects.OrangeHRM;
+using OrangeHRMTestFramework.PageObjects.OrangeHRM.Popups;
+
+namespace OrangeHRMTestFramework.Tests.OrangeHRM
+{
+    public class EmployeeManagementTests : BaseTest
+    {
+        public EmployeeManagementTests() : base(UserInfo.AdminUserInfo)
+        {
+
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            GenericPages.EmployeeManagementPage.ClickLeftNavCategory(LeftNavCategories.Pim);
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            GenericPages.EmployeeManagementPage.ClickTopNavCategoryWithoutSubCategory(PimTopNavCategories.EmployeeList);
+        }
+
+        [Test]
+        public void AddEmployeeTest()
+        {
+            var firstName = RandomHelper.GetRandomString(7);
+            var lastName = RandomHelper.GetRandomString(7);
+            var middleName = RandomHelper.GetRandomString(7);
+
+            // Go to the Add Employee Page using Add Employee button
+            GenericPages.EmployeeManagementPage.ClickAddEmployeeButton();
+
+            // Check that Id input is not empty
+            var addEmployeeForm = GenericPages.AddEmployeePage.AddEmployeeTab;
```

Comment:
Yes, it works, but looks not so good (logical) to specify object of "module of the page" in a test.
As discussed, I prefer to use chain of calls starting from page:
1) var id = GenericPages.AddEmployeePage.AddEmployeeTab.GetValueFromIdTextBox();

or
2) Methods of the page which lead to another page the return type of this page
var addEmployeePage = GenericPages.EmployeeManagementPage.ClickAddEmployeeButton();
var id = addEmployeePage.AddEmployeeTab.GetValueFromIdTextBox();

Reply:
Please check other places in code

Reply:
Yes, your test should start on some page and in this case you need to get object of this page somehow))
Here are 2 ways again:
1. Create object directly in test (e.g. var mainPage = new MainPage()
2. use generic pattern but not for assigning this object for a new variable, just using GenericPage... until any method of the page lead (return) to a new page (e.g.
var userManagementPage = GenericPages.MainPage.OpenUserManagementPage();
var groupPopup = userManagementPage.GroupGrid.OpenEditGroupPopup(_userGroup.Name);)

Reply:
Got it, sorry 🥲 Hopefully, fixed all tests 🤞